api/v1alpha1/webrequestcommitstatus_types.go

```diff
@@ -0,0 +1,275 @@
+	// to determine which environments this validation applies to.
+	// +required
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	Key string `json:"key"`
+
+	// Description is a human-readable description of this validation that will be shown in the SCM provider
+	// (GitHub, GitLab, etc.) as the commit status description.
+	// If not specified, defaults to empty string.
+	// +optional
+	// +kubebuilder:validation:MaxLength=140
+	Description string `json:"description,omitempty"`
+
+	// ReportOn specifies which commit SHA to report the CommitStatus on.
+	// - "proposed": Reports on the proposed hydrated commit SHA (default)
+	// - "active": Reports on the active hydrated commit SHA
+	//
+	// When "proposed": Polls until success, then stops polling for that SHA.
+	// When "active": Polls forever, even after success (active state can change).
+	//
+	// +optional
+	// +kubebuilder:default="proposed"
+	// +kubebuilder:validation:Enum=proposed;active
+	ReportOn string `json:"reportOn,omitempty"`
+
+	// HTTPRequest defines the HTTP request configuration.
+	// Supports Go templates in URL, Headers, and Body fields.
+	// +required
+	HTTPRequest HTTPRequestSpec `json:"httpRequest"`
+
+	// Expression is evaluated using the expr library (github.com/expr-lang/expr) against the HTTP response.
+	// The expression must return a boolean value where true indicates the validation passed.
+	//
+	// Available variables in the expression context:
+	//   - Response.StatusCode (int): the HTTP response status code
+	//   - Response.Body (any): parsed JSON as map[string]any, or raw string if not JSON
+	//   - Response.Headers (map[string][]string): HTTP response headers
+	//
+	// Examples:
+	//   - Response.StatusCode == 200
+	//   - Response.StatusCode == 200 && Response.Body.approved == true
+	//   - Response.StatusCode == 200 && Response.Body.status == "approved"
+	//   - len(Response.Headers["X-Approval"]) > 0
+	//
+	// +required
+	Expression string `json:"expression"`
+
+	// PollingInterval controls how often to retry the HTTP request while in pending state.
+	// When reportOn is "proposed": stops polling after success for a given SHA.
+	// When reportOn is "active": always polls at this interval.
+	// +optional
+	// +kubebuilder:default="1m"
+	PollingInterval metav1.Duration `json:"pollingInterval,omitempty"`
+}
+
+// HTTPRequestSpec defines the HTTP request configuration.
+type HTTPRequestSpec struct {
+	// URL is the HTTP endpoint to request.
+	// Supports Go templates with the following variables:
+	//   - {{ .ProposedHydratedSha }}: the proposed commit SHA
+	//   - {{ .ActiveHydratedSha }}: the active/deployed commit SHA
+	//   - {{ .Key }}: the WebRequestCommitStatus key
+	//   - {{ .Name }}: the WebRequestCommitStatus resource name
+	//   - {{ .Namespace }}: the WebRequestCommitStatus namespace
+	// +required
+	URL string `json:"url"`
```

Comment:
Maybe call it URLTemplate? We could also probably set minLength to 1.

Reply:
templating is available in more than just url, I updated docs to make it more clear url, body, headers